lib/view_model/settings/other_settings_view_model.dart

```diff
@@ -81,13 +94,25 @@ abstract class OtherSettingsViewModelBase with Store {
 
   String getBuyProviderType(dynamic buyProviderType) {
     final _buyProviderType = buyProviderType as BuyProviderType;
+    return _buyProviderType == BuyProviderType.askEachTime
+        ? S.current.ask_each_time
+        : _buyProviderType.name;
+  }
 
-    return _buyProviderType.toString();
+  String getSellProviderType(dynamic sellProviderType) {
+    final _sellProviderType = sellProviderType as BuyProviderType;
+    return _sellProviderType == BuyProviderType.askEachTime
+        ? S.current.ask_each_time
+        : _sellProviderType.name;
   }
 
   void onDisplayPrioritySelected(TransactionPriority priority) =>
       _settingsStore.priority[_wallet.type] = priority;
 
-  void onBuyProviderTypeSelected(BuyProviderType buyProviderType) =>
+  BuyProviderType onBuyProviderTypeSelected(BuyProviderType buyProviderType) =>
       _settingsStore.defaultBuyProviders[walletType] = buyProviderType;
+
+  BuyProviderType onSellProviderTypeSelected(
```

Comment:
these 2 should be annotated with `@action`

Reply:
fixed